bc_obps/service/report_additional_data.py

```diff
@@ -0,0 +1,47 @@
+from registration.models import Operation, RegistrationPurpose
+from reporting.models import ReportVersion, Report, ReportAdditionalData
+from reporting.schema.report_additional_data import ReportAdditionalDataIn
+
+
+class ReportAdditionalDataService:
+    @staticmethod
+    def get_report_person_responsible_by_version_id(
+        report_version_id: int,
+    ) -> ReportAdditionalData:
+        return ReportAdditionalData.objects.get(report_version__id=report_version_id)
+
+    @staticmethod
+    def get_registration_purpose_by_version_id(version_id: int) -> dict:
+        # Fetch the operation associated with the report version
+        operation = Operation.objects.get(
+            id=Report.objects.get(id=ReportVersion.objects.get(id=version_id).report_id).operation_id
+        )
+
+        # Use list comprehension to collect registration purposes directly
+        purposes_list = list(
+            RegistrationPurpose.objects.filter(operation=operation).values_list('registration_purpose', flat=True)
+        )
+
+        # Always return the result as a dictionary with an array (empty or populated)
+        return {"registration_purposes": purposes_list}
+
+    @staticmethod
+    def save_report_additional_data(version_id: int, data: ReportAdditionalDataIn) -> ReportAdditionalData:
+        # Fetch the report version
+        report_version = ReportVersion.objects.filter(id=version_id).first()
+        if report_version is None:
+            raise ValueError("ReportVersion with this ID does not exist.")
```

Comment:
When we know we should get 1 or 0 objects, it's better to use .get rather than .filter().first(). We can replace this bit with a get by pk and allow it to 404 error if no report_version is found rather than throwing our own error here.

```suggestion
        report_version = ReportVersion.objects.get(pk=version_id)
```